src/controllers/controller.bar.js

```diff
@@ -474,16 +468,26 @@ module.exports = function(Chart) {
 		},
 
 		removeHoverStyle: function(rectangle) {
-			var dataset = this.chart.data.datasets[rectangle._datasetIndex];
+			this.updateRectangle(rectangle);
+		},
+
+		updateRectangle: function(rectangle, options) {
```

Comment:
How about calling this method `setDefaultStyles` or something that indicates how the rectangle is being updated

Neither caller of this function passes in the second parameter...

Reply:
I agree `updateRectangle` is not the best, but I would expect `setDefaultStyles` to set all default styles. Is this the case here ? If not, it's a bit misleading. 

As for the second parameter, it can be used by a subclass to supply different default options without having to reimplement the complete method. 
Like this : https://github.com/loicbourgois/chartjs-chart-financial/commit/b5eb15983dcadfcf4f82111a69a712d2578f3c61
It's just an idea, but maybe that's not the best way to do it. 
Also, when trying to set a default value like `(rectangle, options = null)`, the build fails.